### PR TITLE
feat(mongo back): rename step TCTC-1830

### PR DESCRIPTION
**Pull request by raphaelvignes**
* Rename Step in Mongo Back.

### CODE REVIEW

File: server/src/weaverbird/backends/mongo_translator/steps/rename.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from typing import List
+
+from weaverbird.pipeline.steps import RenameStep
+
+
+def translate_rename(step: RenameStep) -> List:
+    return [
+        {'$addFields': {renaming[1]: f'${renaming[0]}'} for renaming in step.to_rename},
```
Comment (davinov) [inline server/src/weaverbird/backends/mongo_translator/steps/rename.py:8]: I think there is a subtle mistake here: the `for` seems outside of the dict comprehension, I think it should be inside.
Check the difference in output:
```python
>>> {'$addFields': {renaming[1]: f'${renaming[0]}'} for renaming in [['a', 'b'], ['c', 'd']]}
{'$addFields': {'d': '$c'}}
>>> {'$addFields': {renaming[1]: f'${renaming[0]}' for renaming in [['a', 'b'], ['c', 'd']]}}
{'$addFields': {'b': '$a', 'd': '$c'}}
```

To be sure not to miss that, what about enriching the test case with multiple columns being renamed at once?

File: server/src/weaverbird/backends/mongo_translator/steps/rename.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+        {'$project': {renaming[0]: 0} for renaming in step.to_rename},
```
Comment (davinov) [inline server/src/weaverbird/backends/mongo_translator/steps/rename.py:9]: Same note as above, with a twist: if we want to rename column A to B and column B to C, then we'll end up with B being removed (only C in the output).
Even if this is a subtle edge case that should honestly not happen a lot, we can manage it quite easily iof we add a condition to the suppression: 
`for renaming in step.to_rename if renaming[1] not in new_column_names` with `new_column_names` being `[renaming[1] for renaming in setp.to_rename]`

**Review by davinov — CHANGES_REQUESTED** [review]
Thanks for this! I may have a few concerns though:

(it was not that easy after all :p)

File: server/src/weaverbird/backends/mongo_translator/steps/rename.py (hunk 1 above)
Comment (raphaelvignes) [reply server/src/weaverbird/backends/mongo_translator/steps/rename.py:8]: Okay, I missed the fact that we could have multiple renamings.

**Review by davinov — APPROVED** [review]

**Pull request merged**